tests/20-SideMenu/PinnedMenu.test.tsx

```diff
@@ -25,81 +28,216 @@ describe("Boards", () => {
+    expect(screen.queryByLabelText("anime has new updates")).not.toBeInTheDocument();
+    expect(within(mutedBoard).queryByRole("presentation", {hidden: true})).not.toBeInTheDocument();
+    const svgs = mutedBoard.getElementsByTagName("svg");
+    expect(svgs).toHaveLength(1);
+    expect(svgs[0]).toHaveClass("fa-volume-mute");
+    expect(svgs[0]).toBeVisible();
   });
 });
 
 describe("Loading", () => {
   test("Renders loading section", async () => {
     render(<Loading />);
   
-    //TODO: fill this
+    expect(screen.getByLabelText(`loading ${(Loading.args as PinnedMenuSectionProps).sectionId}`)).toBeVisible();
+    expect(screen.getAllByLabelText("loading pinned item placeholder")).toHaveLength((Loading.args as LoadingPinnedSectionProps).loadingElementsCount);
   });
 });
 
 describe("Icons", () => {
   test("Renders section with icons", async () => {
     render(<Icons />);
   
-    //TODO: fill this
+    expect(screen.getByLabelText((Icons!.args! as BasePinnedSectionProps).sectionId!)).toBeVisible();
+
+    const argsItems = ((Icons.args as WithPinnedSectionProps)
+      .items as (CircleButtonProps & { id: string, menuOptions?: DropdownProps["options"] })[]);
+
+    const linkIcons = screen.getAllByRole("link");
+    const linkArgs = argsItems.filter(item => !item.menuOptions);
+    expect(linkIcons).toHaveLength(linkArgs.length);
+    linkIcons.forEach((icon, i) => {
+        expect(icon).toHaveAccessibleName(expect.stringContaining(linkArgs[i].id));
+    });
+
+    const buttonIcons = screen.getAllByRole("button");
+    const buttonArgs = argsItems.filter(item => item.menuOptions);
+    expect(buttonIcons).toHaveLength(buttonArgs.length);
+    const buttonArgIndex = argsItems.findIndex(item => item.menuOptions);
+    expect(buttonIcons[0]).toHaveAccessibleName(argsItems[buttonArgIndex].id);
+  });
+
+  test("Icons with links have correct links", async () => {
+    render(<Icons />);
+    const actionReturn = jest.fn();
+    mocked(action).mockReturnValue(actionReturn);
+
+    const argsItems = ((Icons.args as WithPinnedSectionProps)
+    .items as (CircleButtonProps & { id: string, menuOptions?: DropdownProps["options"] })[]);
+
+    const linkArgs = argsItems.filter(item => !item.menuOptions);
+    
+    const iconLinks = screen.getAllByRole("link");
+
+    for (const [i, iconLink] of iconLinks.entries()) {
+      userEvent.click(iconLink);
+      await waitFor(() => {
+        expect(action).toBeCalledWith(`#${linkArgs[i].id}`);
+        expect(iconLink).toHaveAttribute("href", `#${linkArgs[i].id}`);
+      });
+    }
   });
   
+  test("Correctly marks icon without notification", async () => {
+    render(<Icons />);
+  
+    expect(screen.getByRole("link", { name: "heart"})).toBeInTheDocument; 
+    expect(screen.queryByLabelText("heart has new updates")).not.toBeInTheDocument; 
+    expect(document.querySelector(".has-dot")).not.toContainElement(screen.getByRole("link", { name: "heart"}));
```

Comment:
I would suggest doing it the reverse way as the "recommended way". Not sure about the syntax but something like:

`expect(screen.getByRole("link", { name: "heart"}).classList).not.toContain("has-dot")`

Usually I'd err on the side of doing assertions where what's within expect is the element upon testing. Most likely, this would result in clearer error messages if this failed/easier debugging, but also it's generally less cognitive load when this assumption is respected across different tests/statements.

Reply:
I would normally have done it that way, but the `has-dot` class isn't on the link itself, it on the grandparent(?)/ancestor(?) (what's the proper term for parent but two steps away?) `div` and it didn't seem appropriate to add a label to that wrapper `div`. I suppose could add a `data-testid={withNotification ? "notification-dot" : null}`  (would `null` be correct here?) to it and then do:
```
const iconsWithNotificationDots = screen.getAllByTestID("notification-dot");
for (const iconWithNotificationDot of iconsWithNotificationDots) {
  expect(within(iconWithNotificationDot).queryByRole("link", { name: "heart"})).not.toBeInTheDocument();
}
```
if you think that would be easier to read.

Reply:
Mmmh. Let's keep it this way for now, and we can revisit in the future. I'd rather not deal with adding test IDs.

Reply:
That was my feeling as well